spikeinterface/core/waveform_extractor.py

```diff
@@ -289,6 +289,32 @@ def create(cls, recording, sorting, folder, mode="folder", remove_if_exists=Fals
     def is_sparse(self):
         return self.sparsity is not None
 
+    def has_waveforms(self):
+        if self.folder is not None:
+            if self.format == "binary":
+                return (self.folder / "waveforms").is_dir()
+            elif self.format == "zarr":
+                import zarr
+                root = zarr.open(self.folder)
+                return "waveforms" in root.keys()
+        else:
+            return self._memory_objects != {"wfs_arrays": {}, "sampled_indices": {}}
```

Comment:
```suggestion
            return "wfs_arrays" in self._memory_objetcs and "sampled_indices" in self._memory_objetcs
```
equality on dict is dangerous!!